File: Task-03/primes.go
package main

import (
	"fmt"
)

func find_primes(n int) []int {
	arr := make([]bool, n-1)
	for i := 0; i < n-1; i++ {
		arr[i] = true
	}

	p := 2
	for p*p <= n {
		if arr[p-2] {
			for i := p * p; i <= n; i += p {
				arr[i-2] = false
			}
		}
		p++
	}

	var primes []int
	for i := 2; i <= n; i++ {
		if arr[i-2] {
			primes = append(primes, i)
		}
	}

	return primes
}

func main() {
	num := 50 
	primeNumbers := find_primes(num)

	fmt.Printf("Prime numbers up to %d are: %v\n", num, primeNumbers)
}

File: Task-03/primes.js
function find_primes(n) {
    const arr = new Array(n - 1).fill(true);


    for (let p = 2; p * p <= n; p++) {
        if (arr[p - 2]) {
            for (let i = p * p; i <= n; i += p) {
                arr[i - 2] = false;
            }
        }
    }

    var primes = [];
    for (let i = 2; i <= n; i++) {
        if (arr[i - 2]) {
            primes.push(i);
        }
    }

    return primes;
}

var num = 100;
var primeNumbers = find_primes(num);
console.log(`prime numbers less than ${num} are: ${ primeNumbers.join(', ') }`);
